src/app/FakeLib/Sxshelper.fs

```diff
@@ -0,0 +1,354 @@
+﻿/// Module that enables creating and embedding Side-by-Side interop
+/// manifests for registration free deployment of Com-.net interop projects
+module Fake.Sxshelper
+
+open Fake
+open System
+open System.IO
+open System.Linq
+open System.Xml.Linq
+
+/// Represents a `.net` assembly that may be used in COM interop projects` 
+type InteropAssemblyData = 
+    {
+        /// Assembly name
+        Name:string
+        
+        /// Path to the assembly file
+        Path:string 
+        
+        /// Assembly version
+        Version:string 
+        
+        /// Guid from the `System.Runtime.Interopservice.GuidAttribute` of the assembly
+        Guid:System.Guid
+    }
+
+/// Represents an executable to create an _application manifest_ for
+type InteropApplicationData = 
+    {
+        /// Path of the executable binary file
+        ExecutablePath:string
+        
+        /// Version of the executable
+        Version:String
+
+        /// Depandant `.net`assemblies of the executable
+        Dependencies:InteropAssemblyData seq 
+    }
+
+/// Represents status of attempted parsing
+/// of IL file created from executing `ildasm.exe`
+/// on a binary
+type private ILparsingResult =
+    /// Found all required data
+    /// Includes structured assembly data
+    | Success of InteropAssemblyData
+    /// Failed to find all reguired data
+    /// Includes an error message
+    | Failed of string
+
+let private sdkBasePath = ProgramFilesX86 @@ "Microsoft SDKs\Windows"
+let private netFxDir = (Environment.GetEnvironmentVariable("WINDIR")) @@ "Microsoft.NET\Framework"
+
+/// Helper function to help find framework or sdk tools from the 
+/// newest toolkit available
+let private getNewestTool toolPaths = 
+       toolPaths 
+       |> Seq.sortBy (fun p -> p) 
+       |> Array.ofSeq 
+       |> Array.rev 
+       |> Seq.ofArray 
+       |> Seq.head
+
+/// Path to `mt.exe`
+/// ref: https://msdn.microsoft.com/en-us/library/aa375649(v=vs.85).aspx
+let private  mtToolPath = !! (sdkBasePath + "/**/mt.exe") -- (sdkBasePath + "/**/x64/*.*") 
+                          |> getNewestTool
+
+/// Path to `regasm.exe`
+/// Reason for this to be public is that the VB6 helpe uses this
+/// but the `getNewestToll` function is in this module
+let regAsmToolPath = !! (netFxDir + "/**/RegAsm.exe")  
+                             |> getNewestTool
+
+/// Path to `ildasm.exe
+/// .net fx dissasembly tool
+/// ref: https://msdn.microsoft.com/en-us/library/f7dy01k1(v=vs.110).aspx
+let private ildasmPath = !! (sdkBasePath + "/**/ildasm.exe") -- (sdkBasePath + "/**/x64/*.*")
```

Comment:
Might be possible, but there are so many options for the ildasm tool that I wouldn't know how to create something that is reusable for anything useful. Currently I know only this specific usecase. Might be better to wait, and consider adding a generic helper if more usecases would occur.

(To be frank, I would very much like to find some other way of getting the metadata, that does not rely on the porcelain output of ildasm)